File: langgraph/data/equiv/57.py
def check_monotonic_sequence(sequence: list):
    ascending = sequence.copy()
    descending = sequence.copy()
    
    ascending.sort()
    descending.sort(reverse=True)
    
    if sequence == ascending or sequence == descending:
        return True
    return False

# method name: monotonic -> check_monotonic_sequence
# parameter name: l -> sequence
# introduced ascending and descending variables for clarity
# used sort() method on copies of the list instead of sorted() function directly
# kept the same logic but changed the implementation style

File: langgraph/result/qwen/3b-eq2js/57.js
function checkMonotonicSequence(sequence) {
    let ascending = [...sequence];
    let descending = [...sequence];

    ascending.sort((a, b) => a - b);
    descending.sort((a, b) => b - a);

    if (sequence.every((value, index) => value === ascending[index])) {
        return true;
    }
    if (sequence.every((value, index) => value === descending[index])) {
        return true;
    }
    return false;
}

console.log(checkMonotonicSequence([1, 2, 2, 3])); // true
console.log(checkMonotonicSequence([3, 2, 2, 1])); // true
console.log(checkMonotonicSequence([1, 3, 2]));   // false